src/main/java/org/team1540/robot2022/utils/FeatherClient.java

```diff
@@ -35,101 +42,184 @@ public static double getTimer() {
-        }
+        DriverStation.waitForData();
+        String eventName = DriverStation.getEventName();
+        return String.format("%s-%dr%d-%s%d",
+                !Objects.equals(eventName, "") ? eventName : "unknown",
+                DriverStation.getMatchNumber(),
+                DriverStation.getReplayNumber(),
+                DriverStation.getAlliance().toString(),
+                DriverStation.getLocation()
+        );
     }
 
     /**
-     * Record a shot to a local file
+     * Record a shot and result to a local file
      *
-     * @param limelightDistance distance reported by limelight
-     * @param lidarDistance     distance reported by lidar
-     * @param frontRPM          front flywheel RPM
-     * @param rearRPM           rear flywheel RPM
-     * @param hoodUp            is the hood up?
+     * @param shot       shooting parameters
+     * @param firstBall  what happened to the first ball?
+     * @param secondBall what happened to the second ball?
      */
-    private static void recordShotToFile(double limelightDistance, double lidarDistance,
-                                         double frontRPM, double rearRPM,
-                                         boolean hoodUp, Shooter.ShooterProfile profile) {
-        String jsonl = String.format("{\"matchId\": \"%s\", \"timer\": %f, " +
+    private static void commitShot(ShootingParameters shot, ShotResult firstBall, ShotResult secondBall) {
+        String jsonString = String.format("{\"matchId\": \"%s\", \"timer\": %f, " +
                         "\"limelightDistance\": %f, \"lidarDistance\": %f, " +
-                        "\"frontRPM\": %f, \"rearRPM\": %f, \"hoodUp\": %b, \"profile\": \"%s\"}",
-                getMatchID(),
-                getTimer(),
-                limelightDistance, lidarDistance,
-                frontRPM, rearRPM,
-                hoodUp, profile+"");
-        System.out.println(jsonl);
+                        "\"frontRPM\": %f, \"rearRPM\": %f, \"hoodUp\": %b, \"profile\": \"%s\", " +
+                        "\"firstResult\": %s, \"secondResult\": %s}",
+                shot.matchId, lastShot.matchSeconds,
+                shot.limelightDistance, shot.lidarDistance,
+                shot.frontRPM, shot.rearRPM, shot.hoodUp, shot.profile + "",
+                firstBall + "", secondBall + ""
+        );
+
+        // Write shot log to file
         try {
-            Files.writeString(
-                    Paths.get("/home/lvuser/feather.jsonl"),
-                    jsonl + System.lineSeparator(),
-                    StandardOpenOption.APPEND
-            );
+            Files.writeString(Paths.get(shot.directoryPrefix + "/shot.json"), jsonString + System.lineSeparator(), StandardOpenOption.CREATE);
         } catch (IOException e) {
-            DriverStation.reportError("[feather] Unable to append to local file: " + e, true);
+            DriverStation.reportError("[feather] Unable to create shot log file: " + e, true);
```

Comment:
```suggestion
            DriverStation.reportError("[feather] Unable to create shot log file", e.getStackTrace());
```
Might be more informative, I'm not sure what `e` will print as

Reply:
`getStackTrace` returns an array of `StackTraceElement`s, whereas `DriverStation`'s `reportErrorImpl` builds a nicely formatted stack trace as a string so I think the current way will look nicer as an error message.